app/controllers/saml_controller.rb

```diff
@@ -50,27 +47,35 @@ def acs
     @response = OneLogin::RubySaml::Response.new(params[:SAMLResponse], settings: settings)
 
     if @response.is_valid?
-      # TODO params[:SAMLResponse] _should be_ what CMR wants us to pass as a token
-      # However, currently this is causing a ActionDispatch::Cookies::CookieOverflow
-      # problem. We may want to use https://github.com/rails/activerecord-session_store
-      # to store and pass it
-      # TODO since the SAML library is able to decrypt SAMLResponse to data we can use,
-      # can it also re-encrypt it?
-      # session[:launchpad_response_string] = params[:SAMLResponse]
+      session[:launchpad_cookie] = pull_launchpad_cookie
 
       attributes = @response.attributes
       session[:auid] = attributes[:auid]
-      # session[:email] = attributes[:email]
-      # TODO need to verify and set what other session info is needed
+      session[:launchpad_email] = attributes[:email]
+      # session expiration time to require the user to authenticate with Launchpad again
+      # it should be set to 15 min, the default Launchpad session time
+      session[:expires_in] = 900
+      session[:logged_in_at] = Time.now.to_i
+
+      # for now, this requires that the user already has their auid and urs_uid associated in URS
```

Comment:
Is there a ticket that will change this already planned?

Reply:
Yes, MMT-1297. I added the ticket number in a comment elsewhere, but I'll add it here too